discovery-infra/tests/test_auto_role_selection.py

```diff
@@ -7,40 +7,28 @@
 class TestRoleSelection(BaseTest):
     def test_automatic_role_assignment(self, api_client, nodes, cluster):
         """Let the system automatically assign all roles in a satisfying environment."""
-        cluster_id = cluster().id
+        new_cluster = cluster()
 
-        self.setup_hosts(cluster_id=cluster_id,
-                         api_client=api_client,
-                         nodes=nodes)
-        self.set_network_params(cluster_id=cluster_id,
-                                api_client=api_client,
-                                controller=nodes.controller)
+        new_cluster.setup_nodes(nodes)
+        new_cluster.set_network_params(nodes.controller)
+        new_cluster.wait_for_ready_to_install()
+        new_cluster.start_install()
```

Comment:
The start install on Base is also wait_until_cluster_starts_installation.
What happen here? maybe need to add wait for installing in progress?

Reply:
changed